### PR TITLE
Implement Get ETH page

### PR DESCRIPTION
**Pull request by wackerow**
## Description
- Implements Get ETH page as `/src/pages/get-eth.tsx`
- Implements `CentralizedExchanges` component, and associated `useCentralizedExchanges` hook (renamed from `EthExchnges` and `useEthExchanges`)
- Restructured exchange data as a TypeScript object export
- Components associated with `/get-eth` page were also reviewed and updated to latest site standards

## Todo—Migrate/update remaining associated components:
- [x] `EthPriceCard`
- [x] `ButtonLink`
- [x] `FeedbackCard`
- [x] `InfoBanner`

## Related Issue
https://www.notion.so/efdn/Migrate-React-pages-8acee0dc80d04e55b109e9852c20ef2f?pvs=4

## Preview link
https://deploy-preview-152--ethereum-org-fork.netlify.app/en/get-eth
https://deploy-preview-152--ethereum-org-fork.netlify.app/tr/get-eth
https://deploy-preview-152--ethereum-org-fork.netlify.app/zh/get-eth

### CODE REVIEW

File: src/hooks/useCentralizedExchanges.ts [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+  const { locale } = useRouter()
+  const { t } = useTranslation("page-get-eth")
+  const [selectedCountry, setSelectedCountry] = useState<ExchangeByCountryOption | null>()
+
+  const placeholderString = t("page-get-eth-exchanges-search")
+
+  const lastUpdated = getLocaleTimestamp(
+    locale as Lang,
+    // TODO: Set up git log fetch; confirm no SSR issues
+    // data.timestamp.parent.fields.gitLogLatestDate
```
Comment (wackerow) [inline src/hooks/useCentralizedExchanges.ts]: Another TODO worth noting... need to get the last time the exchange list was updated. We were using `git log` before; using `getStaticProps` I believe we should still be able to do this

- [x] Set up `git log` fetch

File: src/components/CentralizedExchanges/index.tsx [hunk 2]
```diff
@@ -97,32 +92,32 @@ const EthExchanges = () => {
...
       </Text>
-      <StyledSelect
+      <Select
         aria-label={t("page-get-eth-exchanges-header")}
         className="react-select-container"
         classNamePrefix="react-select"
-        options={exchangesByCountry}
-        onChange={handleSelectChange}
+        options={selectOptions}
+        onChange={handleSelectChange as any} // TODO: Fix typing
```
Comment (wackerow) [inline src/components/CentralizedExchanges/index.tsx:114]: @pettinarip Ran into type errors here... the prop type for `handleSelectChange` was altered slightly and now it's complaining about the type not matching for `onChange`... it's working as is, but would prefer to not have the use of `any` here

**Review by corwintines — APPROVED** [review]
Nice work @wackerow!

**Comment by wackerow** [comment]
Awesome, thanks for that patch @corwintines! It's not pixel-for-pixel, but it at least resizes appropriately as the screen width scales without weird overflows.
Separately I'll go through and deprecate entirely the use of `resizeImage` in favor of approach described here: <https://nextjs.org/docs/pages/api-reference/components/image#responsive-images>

**Review by corwintines — APPROVED** [review]

File: src/pages/bug-bounty.tsx [hunk 3]
```diff
@@ -418,7 +417,7 @@ const BugBountiesPage = () => {
       link: "https://pegasys.tech/teku",
       image: useColorModeValue(tekuDark, tekuLight),
     },
-  ].map((client) => ({ ...client, image: resizeImage(client.image, 24) }))
+  ].map((client) => ({ ...client, image: client.image }))
```
Comment (pettinarip) [inline src/pages/bug-bounty.tsx]: no need for the map, right?
```suggestion
  ]
```

File: src/pages/bug-bounty.tsx [hunk 4]
```diff
@@ -438,6 +437,14 @@ const BugBountiesPage = () => {
       link: "https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md",
     },
   ]
+
+  const iconImageProps = {
+    sizes: "60px",
```
Comment (pettinarip) [inline src/pages/bug-bounty.tsx]: Since these are going to be fixed width images, there is no need to specify the `sizes` prop.

File: src/pages/bug-bounty.tsx [hunk 5]
```diff
@@ -438,6 +437,14 @@ const BugBountiesPage = () => {
...
+    style: {
+      width: "60px",
+      height: "auto",
+    },
```
Comment (pettinarip) [inline src/pages/bug-bounty.tsx]: We need to pass these props to the `Image` component to tell Next to resize the images accordingly.

In this case, the `width` will be enough as this is a static import and Next already knows the aspect ratio of the image. With the aspect ratio it can automatically calculate the height.

https://nextjs.org/docs/pages/api-reference/components/image#sizes

File: src/pages/get-eth.tsx [hunk 6]
```diff
@@ -0,0 +1,482 @@
...
+  const lastDeployDate = getLastDeployDate()
+
+  return {
+    props: {
+      ...(await serverSideTranslations(locale!, requiredNamespaces)),
+      lastDeployDate,
+      lastDataUpdateDate,
+    },
+  }
+}) satisfies GetStaticProps<SSRConfig>
```
Comment (pettinarip) [inline src/pages/get-eth.tsx]: `lastDeployDate` and `lastDataUpdateDate` are missing from the Props declaration. We are only passing `SSRConfig`.

```ts
type Props = SSRConfig & {
  lastDeployDate..
}
```

File: src/pages/get-eth.tsx [hunk 7]
```diff
@@ -0,0 +1,482 @@
...
+
+const GetEthPage = ({ lastDataUpdateDate }) => {
```
Comment (pettinarip) [inline src/pages/get-eth.tsx]: We can infer the prop types with `InferGetStaticPropsType`

File: src/lib/utils/gh.ts
Comment (pettinarip) [inline src/lib/utils/gh.ts:1]: Nice!

File: src/components/CentralizedExchanges/index.tsx (hunk 2 above)
Comment (pettinarip) [reply src/components/CentralizedExchanges/index.tsx:114]: Not sure if I understand why this had to be changed. Did we need to refactor this logic?

File: src/components/CardList.tsx [hunk 8]
```diff
@@ -43,18 +43,33 @@ const CardContainer = (props: StackProps) => (
...
   return (
-    <CardContainer {...rest}>
-      {image && <Image src={image} alt={alt ?? ""} minW="20px" />}
+    <CardContainer {...props}>
+      {image && (
+        <Image
+          src={image}
+          alt={alt ?? ""}
+          sizes="20px"
+          style={{ width: "20px", height: "auto" }}
```
Comment (pettinarip) [inline src/components/CardList.tsx]: Related to the previous comment about the `Image` component. For fixed and static imported images, we don't need to use the `sizes` prop and we need to tell the width we want to use.

```ts
// this should be enough
<Image src={image} width={20} />
```

Comment (pettinarip) [reply src/components/CardList.tsx]: While testing this I noticed that our `Image` component uses not only the Next Image width but also the Chakra width, `width: var(--eth-sizes-20)` (20rem), and this is not what we want. As a workaround, we can use the Next Image component in the meantime.

File: src/components/Buttons/ButtonLink.tsx [hunk 9]
```diff
@@ -1,14 +1,28 @@
...
-
-const ButtonLink: React.FC<ButtonLinkProps> = (props) => {
-  return <Button as={BaseLink} activeStyle={{}} {...props} />
+const ButtonLink = ({ customEventOptions, ...props }: ButtonLinkProps) => {
+  const matomoEvent = customEventOptions ?? {
+    eventCategory: "Link",
+    eventAction: "clicked",
+    eventName: "ButtonLink clicked",
+    eventValue: (props.to ?? props.href) as string,
+  }
```
Comment (pettinarip) [inline src/components/Buttons/ButtonLink.tsx:14]: Why are we adding this? do we want to keep track of every `ButtonLink` click? I'd lean towards removing this or discuss it with the team before we implement it.

Also, this is out of scope, we are changing behavior of the component.

**Review by pettinarip — CHANGES_REQUESTED** [review]
Good job @wackerow. Overall it looks good on the preview deploy.

I'll push a few commits to fix things related to the use of images & types. Left some comments about it.

On the other hand, I see a lot of tiny refactors and, in some cases, some of them changing the styles and behavior. I'm not against them but lets try to keep them separated to be more organized, minimize potential bugs, and minimize the QA work for the migration.

File: src/components/Buttons/ButtonLink.tsx (hunk 9 above)
Comment (wackerow) [reply src/components/Buttons/ButtonLink.tsx:14]: Notion card: [Add customEventOptions prop back to ButtonLink & Link](https://www.notion.so/efdn/Add-customEventOptions-prop-back-to-ButtonLink-Link-ef2e0044450b4399b681fce5b3ce937d?pvs=4)

Was related to this card

Comment (wackerow) [reply src/components/Buttons/ButtonLink.tsx:14]: This does add a default though; happy to remove this and just have an optional prop that can be passed.

**Comment by wackerow** [comment]
@pettinarip 
> Not sure if I understand why this had to be changed. Did we need to refactor this logic?

The data structure changed a little for the exchanges when converting from `csv` to a `ts` object...

```ts
  exchanges: { [key in ExchangeName]: StringBoolean }
  // became
  exchanges: string[]
```

Just like before we're using the same type for the select options as we are in the change handler.

I tried reverting some of the naming adjustments and state simplification but that didn't seem to change anything. Could refactor the JSON structure to more closely match the `"TRUE"` and `"FALSE"` usage we had before but that seemed kinda silly when not importing from CSV

File: src/components/Buttons/ButtonLink.tsx (hunk 9 above)
Comment (pettinarip) [reply src/components/Buttons/ButtonLink.tsx:14]: Got it, thanks.

We shouldn't need this in `ButtonLink`, as this action should be handled by the underlying `Link` component. The Link component already handles the matomo events, right? that is how we were doing it in the main repo.

**Comment by pettinarip** [comment]
> @pettinarip
> 
> > Not sure if I understand why this had to be changed. Did we need to refactor this logic?
> 
> The data structure changed a little for the exchanges when converting from `csv` to a `ts` object...
> 
> ```ts
>   exchanges: { [key in ExchangeName]: StringBoolean }
>   // became
>   exchanges: string[]
> ```
> 
> Just like before we're using the same type for the select options as we are in the change handler.
> 
> I tried reverting some of the naming adjustments and state simplification but that didn't seem to change anything. Could refactor the JSON structure to more closely match the `"TRUE"` and `"FALSE"` usage we had before but that seemed kinda silly when not importing from CSV

Ahh got it 👍🏼 I'm fine with the current structure. I don't think it is necessary to change it now.
